pkg/service/association.go

```diff
@@ -58,22 +58,29 @@ func isDone(ctx context.Context) bool {
 }
 
 func ensureSetupPfcpAssociation(ctx context.Context, upf *smf_context.UPF, upfStr string) {
-	var alertTime time.Time
+	alertTime := time.Now()
 	alertInterval := smf_context.SMF_Self().AssociationSetupFailedAlertInterval
+	retryInterval := smf_context.SMF_Self().AssociationSetupFailedRetryInterval
 	for {
 		err := setupPfcpAssociation(upf, upfStr)
 		if err == nil {
 			return
 		}
+		logger.AppLog.Errorf("Failed to setup an association with UPF%s, error:%+v", upfStr, err)
 		now := time.Now()
-		if alertTime.IsZero() || now.After(alertTime.Add(alertInterval)) {
-			logger.AppLog.Errorf("Failed to setup an association with UPF%s, error:%+v", upfStr, err)
+		logger.AppLog.Debugf("now %+v, alertTime %+v", now, alertTime)
+		if now.After(alertTime.Add(alertInterval)) {
+			logger.AppLog.Errorf("ALERT for UPF%s", upfStr)
 			alertTime = now
 		}
-
-		if isDone(ctx) {
+		logger.AppLog.Errorf("Wait %+v until next retry attempt", retryInterval)
+		timer := time.After(retryInterval)
```

Comment:
It's better that this line is before `setupPfcpAssociation(upf, upfStr)` since, in ordinary cases, `setupPfcpAssociation(upf, upfStr)` takes around 4sec: pfcp response timeout  x retries.

Reply:
@YoheiKonno , I moved it to be before setupPfcpAssociation and also modified retry debug statement to:

`"Wait %+v (or less) until next retry attempt" ..` because it might wait in `select` less than the amount of retryTime. I also updated log levels accordingly. Is it ok? Thanks.

Reply:
Thanks, LGTM